### PR TITLE
check available space during upload process

### PR DESCRIPTION
**Pull request by kevodwyer**
This is not ideal in the case of multiple uploads
Also handles the case of file copying and directory copying

### CODE REVIEW

File: src/components/filesystem/index.js [hunk 1]
```diff
@@ -270,15 +271,20 @@ module.exports = {
             if (this.isSecretLink)
 		return;
 	    var that = this;
-	    this.context.getSpaceUsage().thenApply(u => that.usageBytes = u.value_0);
+	    this.context.getSpaceUsage().thenApply(u => {
+	        that.usageBytes = u
```
Comment (ianopolous) [inline src/components/filesystem/index.js:275]: do we not need the value_0?

**Comment by ianopolous** [comment]
Have you tested it with >4GiB size files, and/or space left?

File: src/components/filesystem/index.js (hunk 1 above)
Comment (kevodwyer) [reply src/components/filesystem/index.js:275]: No. We call .toString() on the Long object.

Comment (ianopolous) [reply src/components/filesystem/index.js:275]: Wasn't there a reason for it? Maybe to do with handling the 4GiB barrier?

Comment (kevodwyer) [reply src/components/filesystem/index.js:275]: It wasn't consistent in any case (quotaBytes versus usageBytes) . It is now. I have tested it with large numbers.

Comment (ianopolous) [reply src/components/filesystem/index.js:275]: What sizes have you tested it with?

Comment (kevodwyer) [reply src/components/filesystem/index.js:275]: Usage: 31.68 KiB / 200 GiB
Usage: 31.67 KiB / 20000 GiB
Usage: 31.67 KiB / 2000000 GiB

**Comment by kevodwyer** [comment]
> Have you tested it with >4GiB size files, and/or space left?

I used a VM image and the numbers seen right

**Comment by ianopolous** [comment]
We need to test where the quota, the file size, and the (quota - file size) are just below and above 4 GiB.

**Comment by ianopolous** [comment]
Fixes: https://github.com/Peergos/Peergos/issues/748

File: src/components/filesystem/filesystem.html [hunk 2]
```diff
@@ -266,7 +266,7 @@
   <div v-if="progressMonitors.length>0" class="progressholder">
     <progressbar 
         v-for="progress in progressMonitors" v-if="progress.show"
-      v-on:hide-progress="progressMonitors.pop(progress)"
+      v-on:hide-progress="progressMonitors.splice(progressMonitors.indexOf(progress), 1)"
```
Comment (ianopolous) [inline src/components/filesystem/filesystem.html:269]: great catch! Does this fix the random closing of upload progress when concurrently uploading many files?

Comment (kevodwyer) [reply src/components/filesystem/filesystem.html:269]: yes. Note also that uploads are no longer concurrent. This makes it possible to check available space after each upload and also should fix problems whereby the ui would get out of sync

**Comment by kevodwyer** [comment]
> Have you tested it with >4GiB size files, and/or space left?

Yes. Works as expected

**Review by ianopolous — APPROVED** [review]

**Pull request merged**